core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -451,11 +450,30 @@ public void deleteIndexDirectoryUnderLock(Index index, IndexSettings indexSettin
         IOUtils.rm(indexPaths);
         if (indexSettings.hasCustomDataPath()) {
             Path customLocation = resolveIndexCustomLocation(indexSettings);
-            logger.trace("deleting custom index {} directory [{}]", index, customLocation);
-            IOUtils.rm(customLocation);
+            // shared shard data on shared file systems might have already been deleted by another node,
+            // this can only happen if a shared file system index was closed, then deleted on all nodes
+            if (Files.exists(customLocation)) {
```

Comment:
I rather not make this change (and if we do, let's name the method "delete*IfExists). If I understand correctly it's only needed for shadow indices, which we agreed to tackle in another PR.